src/storage.rs

```diff
@@ -168,18 +184,62 @@ impl<'a> Builder {
 
     /// Creates `Storage` based on given configuration
     /// Examples
-    pub fn build(self) -> Storage {
+    pub fn build<T>(self) -> Storage<T> {
         Storage {
             config: self.config,
             ..Default::default()
         }
     }
 
-    /// Sets a string with work dir as pattern for blob naming
-    /// Examples
-    pub fn work_dir<S: Into<&'a str>>(mut self, work_dir: S) -> Self {
-        // @TODO check path
-        self.config.work_dir = work_dir.into().to_string();
+    /// # Description
+    /// Sets a string with work dir as pattern for blob naming.
+    /// If path not exists, Storage will try to create at initialization stage.
+    /// # Panics
+    /// Panics if path is in readonly mode or path isn't a directory
+    /// # Examples
+    /// ```no-run
+    /// let builder = Builder::new().work_dir("/tmp/pearl/");
+    /// ```
+    pub fn work_dir<S: Into<PathBuf>>(mut self, work_dir: S) -> Self {
+        debug!("set work dir");
+        let path: PathBuf = work_dir.into();
+        debug!("check path existence");
+        if !path.exists() {
+            warn!("no such file or directory: {}", path.display());
+        }
+        debug!("load metadata");
+        let metadata = path
+            .metadata()
+            .map_err(|e| error!("metadata load failed: {}", e))
+            .unwrap();
+        let permissions = metadata.permissions();
+        debug!("check is path readonly");
+        debug!("check is path a dir");
+        if permissions.readonly() {
+            error!("{} is in readonly mode", path.display());
+            panic!("work dir must have write permission");
+        } else if !metadata.is_dir() {
+            error!("{} is not a directory", path.display());
+            panic!("work dir must be a directory");
+        }
+        info!("work dir set: {}", path.display());
+        self.config.work_dir = Some(path);
+        self
+    }
+
+    /// # Description
+    /// Sets storage max blob files num
+    /// Must be greater than zero
+    /// Limited by `u32::MAX`
+    pub fn max_blobs_num<U: Into<u32>>(mut self, max_blobs_num: U) -> Self {
```

Comment:
?

Reply:
+1

We should specify a capacity of single blob (not the maximum number of blobs in storage)